core/src/com/unciv/ui/screens/cityscreen/CityConstructionsTable.kt

```diff
@@ -149,7 +149,10 @@ class CityConstructionsTable(private val cityScreen: CityScreen) {
 
     private fun updateButtons(construction: IConstruction?) {
         buyButtonsTable.clear()
-        if (!cityScreen.canCityBeChanged()) return
+        if (!cityScreen.canChangeState) return
+        /** [UniqueType.MayBuyConstructionsInPupets] support - we need a buy button for civs that could buy items in puppets */
+        if (cityScreen.city.isPuppet && !cityScreen.city.civ.hasUnique(UniqueType.MayBuyConstructionsInPupets)) return 
```

Comment:
```suggestion
        if (cityScreen.city.isPuppet && !cityScreen.city.getMatchingUniques(UniqueType.MayBuyConstructionsInPupets).any()) return 
```

That way, it contains the state for conditionals seventhm was talking about, *and* supports city-specific uniques, such as from religions